### PR TITLE
Moves pressure movement to move loops, master plan revealed 

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

I'm dismayed by how pressure movement works currently. Tying it directly to atmos itself is particularly troublesome, since atmos doesn't always fire consistently.

It working based off probabilities also makes it feel kinda crappy. So lets do move loops instead.

They're going to be more expensive just by default, since we're well, making things move more often, but I think smooth movement even under load is worth it.

Oh and I'm adding code that ensures that negative pressure is treated as pressure in the opposite direction, rather then negative force.
Hopefully this makes things a bit nicer.

Pressure resistance and the movement delay min/max may need tweaking (It is hard to test for this sort of thing on a local, live maybe?), but that's ok 

## Why It's Good For The Game

It's all been building up to this baby. Since move loops are quite fast now there's not gonna even gonna be a ton of overhead to it

## Changelog

:cl:
balance: Pressure movement is smoother now. It also is unhooked from atmos processing, so it'll be more consistent. 
/:cl:

### CODE REVIEW

**Comment by Serxule** [comment]
something to cause MORE lag? praise

**Comment by optimumtact** [comment]
fast wind

File: code/__DEFINES/atmospherics/atmos_helpers.dm [hunk 1]
```diff
@@ -3,6 +3,13 @@
 ///(V) is if the share is vertical or not. True or False
 #define CANATMOSPASS(A, O, V) ( A.can_atmos_pass == ATMOS_PASS_PROC ? A.can_atmos_pass(O, V) : ( A.can_atmos_pass == ATMOS_PASS_DENSITY ? !A.density : A.can_atmos_pass ) )
 
+/// Get the amount of force this atom (A) can resist from turf (T)
+/// Allows for proc overrides if the atom does something unique
+#define GETPRESSURERESIST(A, T) ( A.complex_pressure ? A.get_pressure_resistance(T) : A.pressure_resistance )
```
Comment (Kylerace) [inline code/__DEFINES/atmospherics/atmos_helpers.dm:8]: SCREAMING_SNAKE_CASE()

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/__DEFINES/atmospherics/atmos_helpers.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/__DEFINES/atmospherics/atmos_helpers.dm:8]: My ancestors are smiling on my imperial, can you say the same?

**Comment by MrStonedOne** [comment]
>It working based off probabilities also makes it feel kinda crappy. So lets do move loops instead.

working off of probabilities was designed to give meaning to the level of pressure difference.

also i don't think you've ever seen paper spam lag a server because an airlock was opened. (Don't forget about the maptick implications that a subsystem can't take into account when throttling because maptick hasn't happened yet.)

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 2]
```diff
@@ -359,36 +364,111 @@
...
+		return MOVELOOP_SKIP_STEP
+
+	source.direction = our_turf.archived_pressure_direction
+
+/atom/movable/proc/release_pressure(datum/move_loop/source)
+	SIGNAL_HANDLER
+	UnregisterSignal(src, COMSIG_MOVABLE_MOVED)
+	pressure_moving = FALSE
+
+/atom/movable/Moved(atom/OldLoc, Dir)
```
Comment (MrStonedOne) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm:441]: This is a double definition that makes all movements incur an additional proc call overhead.

**Comment by LemonInTheDark** [comment]
> > It working based off probabilities also makes it feel kinda crappy. So lets do move loops instead.
> 
> working off of probabilities was designed to give meaning to the level of pressure difference.
> 
> also i don't think you've ever seen paper spam lag a server because an airlock was opened. (Don't forget about the maptick implications that a subsystem can't take into account when throttling because maptick hasn't happened yet.)

I'm not saying it was designed poorly, fuck no. I understand why what was done was done. I mean it feels crappy in terms of interacting with it as a player, especially since it's tied directly to atmos process rates.

Admittedly I did to some degree eyeball the right delay ranges. It occurs to me I could better match the average experience by basing the new more directly off the old. Need to look into that further

**Comment by MrStonedOne** [comment]
Thinking on it more probability method fixed two seperate issues.

It made paper spam not a lag fest on minor pressure differences.

It resolved airlocks immediately throwing you out of them or otherwise being something you have to fight against, during high atmos tick rates.

While its not "realistic" that you can enter a fully pressurized station from 0psi as long as you fight past the like 1 tile move from space wind, it was suffering to watch people fight against space wind because the mining shuttle launched while they were boarding, or they were one tile to close to the airlocks when the supply shuttle departed.

(or seeing 5 people stuck in the same exposed space tile unable to move out of it while re-slipping on the clowns pda and then cutting themselves on the 50 pieces of glass on the same tile. ok, i lied, that isn't suffering, its why i observe, but still.)

I don't know enough about this system to know how it fairs on either of these counts, but I did figure I'd let you know so you can do more pointed tests.

**Comment by LemonInTheDark** [comment]
Closing this for now, busy with other stuff, and it needs more consideration

**Pull request closed without merging**